File: 1603-design-parking-system/1603-design-parking-system.ts
class ParkingSystem {
    lot: [number, number, number];
    
    constructor(big: number, medium: number, small: number) {
        this.lot = [big, medium, small]
    }

    addCar(carType: number): boolean {
        if (this.lot[carType-1] > 0) {
            this.lot[carType-1]--
            return true
        } else {
            return false
        }
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * var obj = new ParkingSystem(big, medium, small)
 * var param_1 = obj.addCar(carType)
 */

File: 1603-design-parking-system/1603-design-parking-system.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.lot = {
            "1": big,
            "2": medium,
            "3": small
        }

    def addCar(self, carType: int) -> bool:
        if self.lot.get(str(carType)) > 0:
            self.lot[str(carType)] -= 1
            return True
        else:
            return False
        


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)